chef/cookbooks/opendaylight/recipes/default.rb

```diff
@@ -19,22 +19,21 @@
 
 if node[:platform_family] == "suse"
 
-  package "opendaylight-java"
+  package "opendaylight"
 
-  bash "start opendaylight controller" do
+  service "opendaylight" do
+    supports status: true, restart: true
+    action [:enable, :start]
+  end
+
+  bash "install opendaylight features" do
     user "root"
-    code <<-EOF
-      case "$(pidof karaf | wc -w)" in
-      0) echo "Restarting OpenDayLight Karaf"
-         KARAF_REDIRECT=#{node[:opendaylight][:logfile]} /opt/distribution-karaf-0.4.0-Beryllium/bin/start
-         ;;
-      1) echo "Karaf already running. Nothing to do"
-         ;;
-      *) echo "More than one Containers running. Stopping few."
-         kill $(pidof karaf | awk '{print $1}')
-         ;;
-      esac
-    EOF
+    retries 3
```

Comment:
Karaf takes some time to start ssh access. As this runs immediately after odl service start, it fails if I don't wait. I don't know how much time it takes. So three retries with 2 seconds granularity.